Add single chip tt-xla torch model demos

gengelageTT opened the pull request:
### Ticket
N/A

### Problem description
Request for examples of how an end user might use torch through tt-xla to run a model from HF. 

### What's changed
Added resnet50 and bge_m3 demos that run torch models through tt-xla. Replicates the way we run models through tt-xla's `TorchModelTester` (similar to [this example](https://github.com/tenstorrent/tt-xla/blob/main/examples/pytorch/mnist.py)). 

### Checklist
- [x] New/Existing tests provide coverage for changes

nsumrakTT reviewed (approved):
Are you sure that the cats image is licensed so we can include it in our repo?

gengelageTT commented:
@svuckovicTT From what I understand the demos I added call `torch.compile` which is a form of JIT compilation. I added an eager demo for resnet50 which isn't. If I misunderstood and "jit path" means something else let me know.

gengelageTT commented:
Eager demo running in CI here: https://github.com/tenstorrent/tt-forge/actions/runs/17842495671

LPanosTT reviewed (commented):
Could we download this image in the test instead of adding it to the repo? Similar to how it's done in the tt-torch demo? https://github.com/tenstorrent/tt-torch/blob/main/demos/resnet/resnet50_demo.py

gengelageTT commented:
Changed the resnet50 demos to download the image in the test after confirming the image link is in the [allow-list](https://github.com/tenstorrent/github-ci-infra/blob/87aa0a900bafcc3c7c0ce4c920e3118babc1bb90/ansible/playbooks/roles/restricted-proxy/tasks/main.yaml#L238) for civ2.

Demos are re-running in CI here: https://github.com/tenstorrent/tt-forge/actions/runs/17868155495 , https://github.com/tenstorrent/tt-forge/actions/runs/17868160024
